src/www/app/electron_outline_tunnel.ts

```diff
@@ -87,4 +96,24 @@ export class ElectronOutlineTunnel implements Tunnel {
       console.error(`${this.id} status changed to ${status} but no listener set`);
     }
   }
+
+  private async tryInstallServices(): Promise<boolean> {
+    if (!isLinux) {
+      return Promise.resolve(false);
+    }
+    const confirmation = await this.uiService.showConfirmation('outline-services-installation-confirmation');
```

Comment:
UI does not belong here. Think of ElectronOutlineTunnel as part of an SDK that other apps may use.

Instead, propagate a system misconfigured error and handle it in the app code that is calling `start()`
You also need a method to set up the VPN that the app code can call.